controller/konnect/ops/ops_kongvault.go

```diff
@@ -0,0 +1,208 @@
+package ops
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"slices"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+	sdkkonnectops "github.com/Kong/sdk-konnect-go/models/operations"
+	sdkkonnecterrs "github.com/Kong/sdk-konnect-go/models/sdkerrors"
+	"github.com/samber/lo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kong/gateway-operator/controller/konnect/conditions"
+	k8sutils "github.com/kong/gateway-operator/pkg/utils/kubernetes"
+
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+	"github.com/kong/kubernetes-configuration/pkg/metadata"
+)
+
+// TODO: use vault.GetControlPlaneID() after https://github.com/Kong/kubernetes-configuration/pull/77 merged
+func getKongVaultControlPlaneID(vault *configurationv1alpha1.KongVault) string {
+	if vault.Status.Konnect == nil {
+		return ""
+	}
+	return vault.Status.Konnect.ControlPlaneID
+}
+
+func createVault(ctx context.Context, sdk VaultSDK, vault *configurationv1alpha1.KongVault) error {
+	cpID := getKongVaultControlPlaneID(vault)
+	if cpID == "" {
+		return fmt.Errorf(
+			"can't create %T %s without a Konnect ControlPlane ID",
+			vault, client.ObjectKeyFromObject(vault),
+		)
+	}
+
+	vaultInput, err := kongVaultToVaultInput(vault)
+	if err != nil {
+		return fmt.Errorf("failed to convert KongVault to Konnect vault input: %w", err)
+	}
+	resp, err := sdk.CreateVault(ctx, cpID, vaultInput)
+
+	if errWrapped := wrapErrIfKonnectOpFailed(err, CreateOp, vault); errWrapped != nil {
+		k8sutils.SetCondition(
+			k8sutils.NewConditionWithGeneration(
+				conditions.KonnectEntityProgrammedConditionType,
+				metav1.ConditionFalse,
+				"FailedToCreate",
+				errWrapped.Error(),
+				vault.GetGeneration(),
+			),
+			vault,
+		)
```

Comment:
Let's use `SetKonnectEntityProgrammedConditionFalse` as it has been done in https://github.com/Kong/gateway-operator/pull/633

Reply:
Done.